meeting-notes/2016-12-01.md

```diff
@@ -0,0 +1,308 @@
+* [EP](https://github.com/nodejs/node-eps/pull/20)
+* [Roadmap](https://github.com/nodejs/abi-stable-node/issues/18)
+
+
+General roadmap:
+
+1. Stable NAPI for use with native modules. NAN is currently used by many native
+modules; NAPI is addressing weaknesses of NAN abstractions, specifically
+leaking V8 APIs.
+2. Adoption by native modules in the ecosystem.
+3. Adoption by builtin modules in core. These directly use V8 APIs, will
+eventually need to use NAPI.
+
+@rvagg: TJ worked on a stable abstraction for a native ABI, but was too
+difficult because of changes in V8. How confident are we that this ABI will
+actually be stable for a long term, e.g. 12 months?
+
+@mhdawson: Goal has been to prove that current ABI works with 0.10, 0.12, 4.0,
+6.0, and even ChakraCore. Some adjustments were needed, but it wasn’t that
+difficult to hide changes between versions. Proving that this approach works
+
+@rvagg: What about significant future changes in V8? Do we have their buy-in?
+
+@jasnell: Whose responsibility is it to ensure the VM APIs continue to work with
+NAPI? Node’s or the VM owner’s?
+
+@mhdawson: Depends on needs of Node and each VM.
+
+We provide an `Environment*` pointer in each NAPI API, which gives deeper access
+if necessary.
+
+@aruneshchandra: How do we ensure buy-in from VM owners?
+
+@rvagg: If NAPI is accepted downstream VM owners will have to participate.
+
+@s3ththompson: V8 team has reviewed NAPI work and doesn’t have major concerns,
+but does NAPI constrain system design? Does it constrain both/either Node or VM
+owners?
+
+@jasnell: If there were to be a significant change in V8 which would affect NAPI
+would V8 work together with us to address?
+
+@s3ththompson: Hard part is defining an abstraction which provides enough wiggle
+room for innovation.
+
+@rvagg: Would be good to rephrase from “VM Neutrality,” which is somewhat
+controversial, to “ABI Stable Node”, which is what is most important to users.
+
+@mhdawson: Tested some sizeable modules, perf is within reasonable limits. Next
+step is error handling. Also need more feedback from community.
+
+API is C, some VMs use C++. Goal is to ensure C API and ABI are stable. C++
+wrappers/helpers may not be as stable. We’d like to add some common C++-style
+stuff, like `NAN::New`.
+
+Completed:
+
+* Ported all NAN examples.
+* Ported leveldown and nanomsg.
+
+Next:
+
+* Error handling
+* Perf evaluation.
+
+At that point make available to community and seek further feedback.
+
+@mhdawson: A challenge is that now people can use anything in V8. We will need
+to constrain and support only a subset.
+
+That means this won’t work for 100% of modules, i.e. those which do significant
+work specific to V8.
+
+@jasnell: Are there things in Node which should also be part of this API?
+
+@mhdawson: Some libuv abstractions were already in NAN, learning from some of
+those.
+
+Buffer is another example.
+
+@aruneshchandra: Thinking of trying with native modules in CITGM. 
```

Comment:
@aruneshchandra: our plan is to review the native modules in CITGM to validate they don't use anything outside of the existing API.